pkg/builder/backup_builder.go

```diff
@@ -154,13 +154,15 @@ func (b *BackupBuilder) TTL(ttl time.Duration) *BackupBuilder {
 
 // Expiration sets the Backup's expiration.
 func (b *BackupBuilder) Expiration(val time.Time) *BackupBuilder {
-	b.object.Status.Expiration.Time = val
+	expiration := metav1.NewTime(val)
+	b.object.Status.Expiration = &expiration
```

Comment:
you _could_ save yourself a line & variable declaration on all of these by just doing e.g. 
```
b.object.Status.Expiration = &metav1.Time{Time: val}
```
Or defining a helper function somewhere that does the same.

We can merge without the change, though, given it's a nit/tedious - up to you.

Reply:
I wanted to use the factory method over just initialising the struct, in case some implementation changes in the constructor, but maybe it's not that important. I'm not sure what's more idiomatic in Go.

I could create a helper that uses the constructor and returns the pointer form though, to save all those lines. Where do you think that could go?

Reply:
Fair enough. I'd say given that the type and field are exported, either is equally valid -- if the package author didn't want you to use them directly, presumably they wouldn't export them.

Hmm, always challenging to decide where to put little helper functions like this. We do have a `pkg/util/boolptr` which does something similar - could create a similar `timeptr` helper package, or `pkg/util/kube` would probably be fine too

Reply:
Or, you could just implement it as an unexported func once in each package you use it - which looks to only be 2 or 3 places.

Reply:
> Fair enough. I'd say given that the type and field are exported, either is equally valid -- if the package author didn't want you to use them directly, presumably they wouldn't export them.

That's a good point, actually. I'm probably overthinking this and it probably isn't likely that the implementation will change too much since the intention is to just be a wrapper of time.Time. I'll go with initialising the struct directly to avoid having to find somewhere to put a helper. Thanks!